src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentMarkupBlockPass.cs

```diff
@@ -156,6 +156,20 @@ public override void VisitMarkupElement(MarkupElementIntermediateNode node)
                     // later.
                     _foundNonHtml = true;
                 }
+                else if (string.Equals("option", node.TagName, StringComparison.OrdinalIgnoreCase))
+                {
+                    // Also, treat <option>...</option> as non-HTML - we don't want it to be coalesced so that we can support setting "selected" attribute on it.
+                    // We only care about option tags that are nested under a select tag.
+                    foreach (var ancestor in Ancestors)
+                    {
+                        if (ancestor is MarkupElementIntermediateNode element &&
+                            string.Equals("select", element.TagName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            _foundNonHtml = true;
+                            break;
+                        }
```

Comment:
This seems like an odd implementation choice. What if we just bail on a `select` like we do for `script`?

Reply:
I tried exactly that at first. If we do that it still coalesces the option tags under select.

Reply:
did you return from the method instead of allowing recursion to continue?

Reply:
I think I followed exactly what we did for script.

Reply:
Will we ever have other children of select that we might want to coalesce? If we don't recurse in to select, that will never get coalesced. That's probably fine but want to confirm.

Reply:
I'm not sure really. I think it's fine either way.